CMakeLists.txt

```diff
@@ -20,9 +20,13 @@ project(
   LANGUAGES C CXX
   VERSION ${MLX_PROJECT_VERSION})
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+  add_compile_options(-Wall -Wextra)
```

Comment:
This should be moved to the `# ---- Lib ----` section below, and we should avoid global compile options when possible:

```cmake
target_compile_options(mlx PUBLIC -Wall -Wextra)
```